File: 145. Binary Tree Postorder Traversal.py
class Solution:
    def postorderTraversal(self, root):

        result = []

        def postorder(node):

            # Base case
            if not node:
                return

            # Left
            postorder(node.left)

            # Right
            postorder(node.right)

            # Root
            result.append(node.val)

        postorder(root)

        return result

File: 145. Binary Tree Postorder Traversal.cpp
class Solution {
public:
    vector<int> result;

    void postorder(TreeNode* root) {

        // Base case
        if (root == nullptr) {
            return;
        }

        // Traverse left subtree
        postorder(root->left);

        // Traverse right subtree
        postorder(root->right);

        // Visit current node
        result.push_back(root->val);
    }

    vector<int> postorderTraversal(TreeNode* root) {

        postorder(root);

        return result;
    }
};
